plugins/tpm/fu-self-test.c

```diff
@@ -84,15 +88,15 @@ fu_tpm_device_2_0_func(void)
 	if (tpm_server_running == NULL && (getuid() != 0 || geteuid() != 0)) {
 		g_test_skip("TPM2.0 tests require simulated TPM2.0 running or need root access "
 			    "with physical TPM");
-		return;
+		goto done;
```

Comment:
Will do, I was trying to follow the [single-path cleanup](https://developer.gnome.org/GSettings/documentation/guidelines/programming/memory-management.html#single-path-cleanup) pattern to avoid precisely the kind of bug that occurred here — unfortunately, GLib doesn't seem to have an RAII-like cleanup helper.

Reply:
Done.